x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/AzureOpenAiSecretSettings.java

```diff
@@ -25,12 +25,16 @@
 import static org.elasticsearch.core.Strings.format;
 import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalSecureString;
 
-public record AzureOpenAiSecretSettings(@Nullable SecureString apiKey, @Nullable SecureString entraId) implements SecretSettings {
+public class AzureOpenAiSecretSettings implements SecretSettings {
```

Comment:
This change was needed to correctly test `testDecorateWithAuthHeader_entraIdAndApiKeyMissing_throwMissingAuthValidationException` inside `AzureOpenAiRequestTests`. Mockito cannot mock/spy final classes. IMO this is acceptable for test coverage reasons.